### PR TITLE
TFTRT: Set engine output datatypes based on what TF is expecting

### PR DESCRIPTION
**Pull request by trevor-m**
TRT requires the datatype of engine output tensors to be set explicitly. We will now tell TRT to output in the type that tensorflow is expecting.

### CODE REVIEW

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc [hunk 1]
```diff
@@ -974,9 +976,12 @@ Status Converter::RenameAndMarkOutputTensors(
       MarkQuantizationRangesAsInferrable(tensor, layer->getOutput(0));
       tensor = layer->getOutput(0);
     }
-    tensor->setName(output.second.c_str());
-    VLOG(1) << "Marking output tensor " << output.first << ", as output tensor "
-            << output.second;
+    tensor->setName(output.dest_node_name.c_str());
+    // Set type after marking as output. TRT only supports setType for engine
```
Comment (aaroey) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc]: Do we need to set the type ***after*** marking it as output? Can we add a test for this fix as well?

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes.h [hunk 2]
```diff
@@ -548,6 +546,19 @@ class Converter {
   friend class OpConverterTest;
 };
 
+// Used for Converter::RenameAndMarkOutputTensors()
+struct EngineOutputInfo {
```
Comment (aaroey) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes.h]: Please inline the definition.

**Review by aaroey — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc (hunk 1 above)
Comment (trevor-m) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc]: Yes, it must be after marking it as an output.
I think the only difference is that if you set the type before, TRT will log a warning. But they claim to only support setType for inputs and outputs so we shouldn't do that.

Comment (aaroey) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc]: Cool, thanks. If you can help to add a simple test using maybe TopK, to reproduce the problem (i.e. output dtype is kFLOAT before but kINT32 after), then we're good to go.

Comment (trevor-m) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc]: @aaroey Thanks, I have added a test.

**Review by aaroey — APPROVED** [review]
Thanks for adding the test!

**Pull request merged**